pallets/corporate-actions/src/lib.rs

```diff
@@ -767,6 +709,55 @@ decl_module! {
             CorporateActions::insert(ca_id.ticker, ca_id.local_id, ca.clone());
             Self::deposit_event(Event::RecordDateChanged(agent, ca_id, ca));
         }
+
+         #[weight = <T as Config>::WeightInfo::initiate_corporate_action_use_defaults(
+                T::MaxDidWhts::get(),
+                T::MaxTargetIds::get(),
+            )
+            .max(<T as Config>::WeightInfo::initiate_corporate_action_provided(
+                withholding_tax.as_ref().map_or(0, |whts| whts.len() as u32),
+                targets.as_ref().map_or(0, |t| t.identities.len() as u32),
+            )) + <T as Config>::DistWeightInfo::distribute()
+        ]
+        pub fn initiate_corporate_action_and_distribute(
```

Comment:
I think the tests build is failing because of the large number of parameters here.  Might have to group parameters into a struct.

Reply:
Yep.